From fb81f9dadf9629f4e0a65de566e04e9591e837a1 Mon Sep 17 00:00:00 2001
From: Mikhail Golubev-Ciuchea <mgolubev@qti.qualcomm.com>
Date: Mon, 9 Sep 2024 17:25:47 +0200
Subject: [PATCH] Revert 523796808ae9de6f8fc778a2f5599a8314989189 "vrend: cache
 GL_DRAW_INDIRECT_BUFFER binding"

Revert the commit to avoid skipping of indirect buffer binding when a
buffer is unbound in vrend_create_buffer().

Signed-off-by: Mikhail Golubev-Ciuchea <mgolubev@qti.qualcomm.com>
---
 src/vrend_renderer.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/src/vrend_renderer.c b/src/vrend_renderer.c
index 8cd20b46..76f7fe81 100644
--- a/src/vrend_renderer.c
+++ b/src/vrend_renderer.c
@@ -770,8 +770,6 @@ struct vrend_sub_context {
 
    int last_shader_idx;
 
-   GLint draw_indirect_buffer;
-
    GLint draw_indirect_params_buffer;
 
    struct pipe_rasterizer_state hw_rs_state;
@@ -6047,11 +6045,10 @@ int vrend_draw_vbo(struct vrend_context *ctx,
    }
 
    if (has_feature(feat_indirect_draw)) {
-      GLint buf = indirect_res ? indirect_res->gl_id : 0;
-      if (sub_ctx->draw_indirect_buffer != buf) {
-         glBindBuffer(GL_DRAW_INDIRECT_BUFFER, buf);
-         sub_ctx->draw_indirect_buffer = buf;
-      }
+      if (indirect_res)
+         glBindBuffer(GL_DRAW_INDIRECT_BUFFER, indirect_res->gl_id);
+      else
+         glBindBuffer(GL_DRAW_INDIRECT_BUFFER, 0);
 
       if (has_feature(feat_indirect_params)) {
          GLint buf = indirect_params_res ? indirect_params_res->gl_id : 0;
-- 
2.43.2

